src/main/java/com/fasterxml/jackson/databind/deser/impl/JDKValueInstantiators.java

```diff
@@ -23,19 +24,28 @@ public static ValueInstantiator findStdValueInstantiator(DeserializationConfig c
             Class<?> raw)
     {
         if (raw == JsonLocation.class) {
-            return new JsonLocationInstantiator();
+            return JsonLocationInstantiator.INSTANCE;
         }
         // [databind#1868]: empty List/Set/Map
         // [databind#2416]: optimize commonly needed default creators
         if (Collection.class.isAssignableFrom(raw)) {
             if (raw == ArrayList.class) {
                 return ArrayListInstantiator.INSTANCE;
             }
+            if (raw == LinkedList.class) {
+                return LinkedListInstantiator.INSTANCE;
+            }
+            if (raw == HashSet.class) {
+                return HashSetInstantiator.INSTANCE;
+            }
+            if (raw == TreeSet.class) {
+                return TreeSetInstantiator.INSTANCE;
+            }
             if (Collections.EMPTY_SET.getClass() == raw) {
-                return new ConstantValueInstantiator(Collections.EMPTY_SET);
+                return ConstantValueInstantiator.EMPTY_SET_INSTANCE;
```

Comment:
No, I think these should not be pre-created since they are relatively rarely needed.
Let's leave them dynamically created on as-needed basis.

Reply:
Are they cached somewhere, or will they be instantiated for each deserialization call? I looked around, and it looked like the latter, which is why I made them all like that. If they are cached then it'd make sense to create them as we go and let cache handle the rest.

Reply:
They are actually cached as part of `JsonDeserializer` created.

I think eager creation is fine for default fallback types (`ArrayList`, `LinkedHashMap`, `HashSet`?).
Fine to also use for `JsonLocation`. These are more speculative.

Reply:
I gave it another spin. Default fallbacks remained eager, the rest moved to lazy. Seems like a sweet spot between eager and `new`.

Reply:
Ah. I should have been more careful here; by lazily I just meant creating instance directly with `new` -- effective caching is, as I said, indirect: actual `JsonDeserializer` created for specific type will use `ValueInstantiator`.

But I can actually make changes before merging so I think this is fine.
(also, due to the way class loaders work, I think `final static` instance fields are actually only constructed when class is initialized which does not occur -- AFAIK -- before "substantial" access (or whatever term was used), and not simply because class is referenced).

All I need now is the CLA, as per my note further down. Once that is done I can do whatever mas